app/controllers/whitelist_controller.rb

```diff
@@ -0,0 +1,77 @@
+class WhitelistController < ApplicationController
+
+  # http_basic_authenticate_with name: "cs169", password: ENV['PROJECTSCOPE_PASSWORD']
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  ADMIN_ROLE = "admin"
+  COACH_ROLE = "coach"
+  
+  # GET /whitelist
+  def index
+    if current_user.is_admin?
+       @permitted_users = Authorized_user.all 
+      
+    else
+       flash[:notice] = "You are not authorized to manipulate whitelist."
+       redirect_to projects_url
+    end
+  end
+
+  # GET /whitelist/index
+  def show
```

Comment:
hmm show should be corresponding to some particular whitelist? why does this part redirect to /whitelist/index?